### PR TITLE
Pie port

### PR DESCRIPTION
**Pull request by shepheb**

### CODE REVIEW

File: core/mlang2.js [hunk 1]
```diff
@@ -711,7 +711,7 @@ CLASS({
     reduceI: function(other) {
       for ( var i in other.groups ) {
         if ( this.groups[i] ) this.groups[i].reduceI(other.groups[i]);
-        else this.groups[i] = other.groups[i].deepClone();
```
Comment (adamvy) [inline core/mlang2.js]: Shouldn't this still be deepClone()?  reduceI should merge the results from both the other expression into this one.

Comment (shepheb) [inline core/mlang2.js]: Probably, yes. Fixed.

File: core/mlang2.js [hunk 2]
```diff
@@ -956,9 +964,13 @@ CLASS({
       var acc = this.arg2;
       acc.put(val);
     },
-    clone: function() {
+    exprClone: function() {
       // Don't use default clone because we don't want to copy 'groups'
-      return MapExpr.create({arg1: this.arg1, arg2: this.arg2.clone()});
+      return MapExpr.create({
+        arg1: this.arg1,
+        arg2: Array.isArray(this.arg2) ? this.arg2.sink.exprClone() :
```
Comment (adamvy) [inline core/mlang2.js]: Why the check for Array.isArray?  We shouldn't need to change the mode of arg2.  It's possible that the user wanted an array dao here.

Comment (shepheb) [inline core/mlang2.js]: Yeah, he might well, but that's why the `.sink` is here. Otherwise it results in an [].dao (the default) and then put() is broken.

Comment (adamvy) [inline core/mlang2.js]: If the user puts [].dao as arg2 and you turn it into [].sink on clone then you're misrepresenting what the user wanted.

Comment (adamvy) [inline core/mlang2.js]: If you want to enforce that arg2 only be [].sink, then this is entirely the wrong place to do this, on .exprClone of a single mlang.

File: core/mlang2.js [hunk 3]
```diff
@@ -1027,6 +1043,7 @@ CLASS({
   properties: [
     {
       name: 'value',
+      compareProperty: function() { return 0; },
```
Comment (adamvy) [inline core/mlang2.js]: Why is this necessary?
